tests/python/frontend/caffe/test_forward.py

```diff
@@ -0,0 +1,1003 @@
+    _save_prototxt(n_netspec, proto_file)
+    _save_solver(solver_file, proto_file, blob_file)
+    _save_caffemodel(solver_file, blob_file)
+
+
+def _siso_op(data, func, *args, **kwargs):
+    """ Create single input and single output Caffe op """
+    n = caffe.NetSpec()
+    n.data = L.Input(input_param={'shape': {'dim': list(data.shape)}})
+    n.output = func(n.data, *args, **kwargs)
+    return n
+
+
+def _miso_op(data_list, func, *args, **kwargs):
+    """ Create multi input and single output Caffe op """
+    n = caffe.NetSpec()
+    if not isinstance(data_list, (tuple, list)):
+        raise TypeError("Need tuple or list but get {}".format(
+            type(data_list)))
+    input_list = list()
+    for idx, data in enumerate(data_list):
+        n['data' +
+          str(idx)] = L.Input(input_param={'shape': {
+              'dim': list(data.shape)
+          }})
+        input_list.append(n['data' + str(idx)])
+    n.output = func(*input_list, *args, **kwargs)
+    return n
+
+
+def _simo_op(data, func, *args, **kwargs):
+    """ Create single input and multi output Caffe op """
+    n = caffe.NetSpec()
+    n.data = L.Input(input_param={'shape': {'dim': list(data.shape)}})
+    output_list = func(n.data, *args, **kwargs)
+    for idx, out in enumerate(output_list):
+        n['output' + str(idx)] = out
+    return n
+
+
+def _run_caffe(data, proto_file, blob_file):
+    """ Run caffe model by Caffe according to .caffemodel and .prototxt"""
+    net = caffe.Net(proto_file, blob_file, caffe.TEST)
+    if isinstance(data, (list, tuple)):
+        for idx, d in enumerate(data):
+            net.blobs['data' + str(idx)].data[...] = d
+    else:
+        net.blobs['data'].data[...] = data
+    out = net.forward()
+
+    caffe_output = list()
+    for i in range(len(out.keys())):
+        if 'output'+str(i) not in out.keys():
+            caffe_output.clear()
+            return list(out.values())
+        caffe_output.append(out['output'+str(i)])
+    return caffe_output
+
+
+def _run_tvm(data, proto_file, blob_file):
+    """ Run caffe model by TVM according to .caffemodel and .prototxt"""
+    init_net = pb.NetParameter()
+    predict_net = pb.NetParameter()
+
+    # load model
+    with open(proto_file, 'r') as f:
+        text_format.Merge(f.read(), predict_net)
+    # load blob
+    with open(blob_file, 'rb') as f:
+        init_net.ParseFromString(f.read())
+
+    shape_dict = dict()
+    dtype_dict = dict()
+    if isinstance(data, (tuple, list)):
+        for idx, d in enumerate(data):
+            shape_dict['data' + str(idx)] = d.shape
+            dtype_dict['data' + str(idx)] = 'float32'
+    else:
+        shape_dict = {'data': data.shape}
+        dtype_dict = {'data': 'float32'}
+
+    mod, params = relay.frontend.from_caffe(
+        init_net, predict_net, shape_dict, dtype_dict)
+
+    target = 'llvm'
+    target_host = 'llvm'
+
+    ctx = tvm.cpu(0)
+    with tvm.transform.PassContext(opt_level=2):
```

Comment:
Why can not be opt level 3?

Reply:
> Why can not be opt level 3?

Sorry，this is a historical issue. In previous version, it will throw errors while building LRN layer(Alexnet) with opt_level 3. And this problem has been solved, so i will reset the opt_level to 3. Thanks